flink-end-to-end-tests/test-scripts/test_kubernetes_session.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+################################################################################
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+source "$(dirname "$0")"/common.sh
+source "$(dirname "$0")"/common_kubernetes.sh
+
+CLUSTER_ROLE_BINDING="flink-role-binding-default"
+CLUSTER_ID="flink-native-k8s-session-1"
+FLINK_IMAGE_NAME="test_kubernetes_session"
+LOCAL_OUTPUT_PATH="${TEST_DATA_DIR}/out/wc_out"
+OUTPUT_PATH="/tmp/wc_out"
+ARGS="--output ${OUTPUT_PATH}"
+
+function cleanup {
+    kubectl delete service ${CLUSTER_ID}
+    kubectl delete clusterrolebinding ${CLUSTER_ROLE_BINDING}
+}
+
+start_kubernetes
+
+cd "$DOCKER_MODULE_DIR"
+# Build a Flink image without any user jars
+./build.sh --from-local-dist --job-artifacts ${TEST_INFRA_DIR}/test-data/words --image-name ${FLINK_IMAGE_NAME}
+
+kubectl create clusterrolebinding ${CLUSTER_ROLE_BINDING} --clusterrole=edit --serviceaccount=default:default --namespace=default
+
+mkdir -p "$(dirname $LOCAL_OUTPUT_PATH)"
+
+"$FLINK_DIR"/bin/kubernetes-session.sh -Dkubernetes.cluster-id=${CLUSTER_ID} \
+    -Dkubernetes.container.image=${FLINK_IMAGE_NAME} \
+    -Djobmanager.heap.size=512m -Dtaskmanager.memory.total-process.size=512m \
+    -Dtaskmanager.memory.framework.heap.size=32m -Dtaskmanager.memory.framework.off-heap.size=32m \
+    -Dkubernetes.jobmanager.cpu=0.5 -Dkubernetes.taskmanager.cpu=0.5 \
+    -Dcontainerized.heap-cutoff-min=100
```

Comment:
I am curious about how to consider the setting of `containerized.heap-cutoff-min`, `taskmanager.memory.framework.heap.size`, `taskmanager.memory.framework.off-heap.size`. If we use the default value, are there any concerns here?

Reply:
We just set the memory small here. This is the only reason. Of course, we could use the default value. That means we need to a start big minikube, about <4G, 2Cpu>. I will have a try whether we could start such a minikube on travis.

Reply:
The minikube start with <2048m, 2cpu> by default. The allocatable memory and cpu will be smaller than <2048, 2cpu>. It is not enough for the Flink default memory config options. So do you think we need to start a bigger minikube cluster or use `-D` to lower the flink job memory?

Reply:
I guess the reason is for minimizing the required resources for cluster. The overall cpu and processor memory make sense for me. I only concern of the above three mentioned options which seem a bit specific and hardly to say the usages clearly. 

E.g. we changed the `taskmanager.memory.framework.off-heap.size` from default 128m to 32m.

Another small consideration here is that if the key `taskmanager.memory.framework.off-heap.size` is changed future, this shell also needs to be adjusted accordingly. But if we do not touch it here to use the default value, then we can avoid this potential effect.

Reply:
Thanks a lot. I will take a look again so that we just need to update the config options as few as possible.

Reply:
I will just leave the following dynamical properties to minimalize the flink cluster resource. Does it make sense?
```
-Djobmanager.heap.size=512m \
-Dcontainerized.heap-cutoff-min=100 \
-Dkubernetes.jobmanager.cpu=0.5 \
-Dkubernetes.taskmanager.cpu=0.5 \
```

Reply:
It seems better than before.